src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -859,6 +859,8 @@ func (s *commitLogSource) startM3TSZEncodingWorker(
 			wroteExisting  = false
 		)
 		for i := range unmergedBlock {
+			// TODO(r): Write unit test to ensure that different values that arrive
```

Comment:
This is a good point, did you handle this anywhere? I see a TODO for a unit test, but I don't see the logic at all.

Reply:
I added a link to issue I created:
https://github.com/m3db/m3/issues/898